### PR TITLE
Reapply "feat: SDK 2.0 Support (#31)"

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
This reverts commit 687802f98806708834ce4a12d5d70d1a9b1fc33e. To be merged into `main` only after a patch release containing #32 is released.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -16,7 +16,7 @@ classifiers =
 py_modules = pytest_sentry
 install_requires =
     pytest
-    sentry-sdk
+    sentry-sdk==2.0.0rc4
```
Comment (asottile-sentry) [inline setup.cfg:19]: does this work with `>=2.0.0rc4` ?

Comment (antonpirker) [reply setup.cfg:19]: This version will work with sentry-sdk>=2.0.0 including all the alpha (2.0.0a1, 2.0.0a2) and release candidate versions (2.0.0rc1-2.0.0rc4). I tried to match all this in the other PR but could not get it right (see commits about matching here: https://github.com/getsentry/pytest-sentry/pull/31)

Can you show us how to match all those versions correctly?

Comment (szokeasaurusrex) [reply setup.cfg:19]: Okay, we have fixed it now

**Review by antonpirker — APPROVED** [review]
Looks good!

File: setup.cfg [hunk 2]
```diff
@@ -16,7 +16,7 @@ classifiers =
 py_modules = pytest_sentry
 install_requires =
     pytest
-    sentry-sdk
+    sentry-sdk>=2.0.0a1,<3
```
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply setup.cfg:19]: Noted. #34

**Pull request merged**